### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by mrprim**
Added hooks for Column and Cell definitions.  Added loading logic.

### CODE REVIEW

File: src/components/ButtonPager/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import * as pagerLogic from '../../utils/pagerLogic'
+import PagerButton from '../PagerButton'
+import usePager from '../../hooks/usePager'
+
+const isFirstPage = currentPage => currentPage <= 1
+const isLastPage = (currentPage, totalPages) => currentPage === totalPages
+
+const Pager = ({ className }) => {
+  const { page, pageSize, count, totalPages } = usePager()
+  const numberedPageButtons = pagerLogic.determinePages(page, pageSize, count)
```
Comment (frodare) [inline src/components/ButtonPager/index.js]: this logic should be moved into `usePager`

File: src/components/ButtonPager/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import * as pagerLogic from '../../utils/pagerLogic'
+import PagerButton from '../PagerButton'
```
Comment (frodare) [inline src/components/ButtonPager/index.js]: I think PageButton should be moved into this component folder

File: src/components/DefaultManifestTable/index.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Table from '../Table'
+import Controls from '../Controls'
+import useManifest from '../../hooks/useManifest'
+
+const DefaultManifestTable = ({ className }) => {
+  const { definition, rows, loadingData } = useManifest()
```
Comment (frodare) [inline src/components/DefaultManifestTable/index.js:8]: we should consider having three loading flags, `loading`, `loadingData` (or maybe `loadingPage`), and `loadingCount`. were loading = loadingPage && loadingCount.  And `loading` should be used in this component

File: src/components/Table/index.js [hunk 4]
```diff
@@ -1,23 +1,18 @@
...
-  <div className={`manifest-table ${className || ''}`}>
-    <table>
-      <Headers />
-      <Rows />
-      <tfoot />
-    </table>
-    <Controls />
-    <Debug />
-  </div>
+const Table = ({ className, columnCount, rowCount }) =>
```
Comment (frodare) [inline src/components/Table/index.js]: I would like to see all code dealing with the HTML table in this component. I think we should dissolve the Headers and Rows component into here and this component handle the entire scaffolding of the manifest.

File: src/components/PageSizer/index.js [hunk 5]
```diff
@@ -1,10 +1,8 @@
...
+import usePager from '../../hooks/usePager'
 
-const PagerSizer = ({ filter, changePageSize, loading }) => {
-  const { meta, setPageSize } = useManifestData()
-  const { pageSize } = meta
+export default () => {
+  const { pageSize, setPageSize } = usePager()
   const handlePageSizeChange = ev => setPageSize(+ev.target.value)
 
   return (
```
Comment (frodare) [inline src/components/PageSizer/index.js:8]: I think we should remove all of the bootstrap classes

File: src/hooks/usePager.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { useContext, useCallback } from 'react'
+import { manifestContext } from '../components/Manifest'
+import * as pagerLogic from '../utils/pagerLogic'
+import * as actions from '../actions'
+
+export default () => {
+  const { state, dispatch } = useContext(manifestContext)
```
Comment (frodare) [inline src/hooks/usePager.js]: I think we should only use `useContext` inside of `useManifest`

File: src/utils/reducer.js [hunk 7]
```diff
@@ -1,11 +1,11 @@
 import * as actionTypes from '../constants/actionTypes'
 
 export const initialState = {
```
Comment (frodare) [inline src/utils/reducer.js:3]: after looking over this reducer, I think it should be replaced with some `useState` hooks.  There isn't any complex side effects from setting data, it looks like simple persistence.

File: src/utils/createSequentialIntegerArray.js [hunk 8]
```diff
@@ -0,0 +1 @@
+export default length => Array(length).fill().map((_, i) => i)
```
Comment (frodare) [inline src/utils/createSequentialIntegerArray.js]: I have to say, with a name that like, ... I expected more.

File: src/hooks/usePager.js [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  const { page, pageSize, count } = state
```
Comment (frodare) [inline src/hooks/usePager.js]: after switching to useManifest, we shouldn't need all of these double unpacking lines

**Review by frodare — CHANGES_REQUESTED** [review]

File: src/hooks/usePager.js [hunk 10]
```diff
@@ -1,14 +1,15 @@
 import useManifest from './useManifest'
 import * as pagerLogic from '../utils/pagerLogic'
 
-export default () => {
+export default ({ numberOfPages }) => {
   const { loading, page, pageSize, count, setPage, setPageSize } = useManifest()
   const totalPages = pagerLogic.determineTotalPages(pageSize, count)
-
+  const pages = pagerLogic.determinePages(numberOfPages, page, pageSize, count)
```
Comment (frodare) [inline src/hooks/usePager.js]: we should probably take this in as an object so that the arguments are named

**Review by frodare — APPROVED** [review]

**Review by zeriley — APPROVED** [review]

**Pull request merged**